src/ray/raylet/worker_pool.cc

```diff
@@ -52,7 +53,9 @@ uint32_t WorkerPool::Size() const {
 
 void WorkerPool::StartWorker(bool force_start) {
   RAY_CHECK(!worker_command_.empty()) << "No worker command provided";
-  if (!started_worker_pids_.empty() && !force_start) {
+  // The first condition makes sure that we are always starting up to
+  // num_cpus_ number of processes in parallel.
+  if (NumStartedWorkers() > num_cpus_ && !force_start) {
```

Comment:
Can you update the documentation for `StartWorker` to account for the new logic?

Reply:
Yeah, the documentation should also make clear that `StartWorker` may not actually start a worker.